### ISSUE TITLE
How to reset atom to loading state after reseting an error boundary

### ISSUE THREAD

**Issue by sandrosc** [jotaijs/jotai-tanstack-query#32]
So far I've tried using the `QueryErrorResetBoundary` (like described [here](https://tanstack.com/query/latest/docs/react/reference/QueryErrorResetBoundary)) and calling `queryClient.invalidateQueries(...)` manually when reseting the error boundary, but the outcome is as follows:

- I shut down my server & reload
- The query fetches, fails, component enters error state, gets caught in error boundary
- I click the fallback "try again" button to reset the error boundary
- The component immediately renders in an error state and fails again
- The query refetches, but doesn't seem to update the atom
- Even after clicking "try again" again the same happens

The [error handling example you provide](https://codesandbox.io/s/joer59) seems to work by changing the user state to a different view where the query doesn't fail. When navigating to the item that causes the error, it immediately fails again. How could we reset the query, so instead of failing the atom tries again and reenters the loading state?

**Comment by dai-shi** [comment]
I'd admit the error boundary support is not ideal.

If you are using `dataAtom` (the first one from atomsWithQuery), it works like Jotai atoms and not like TanStack Query.
If you are using `statusAtom` (the second one), it works more like bare TanStack Query, but no error boundary support. (because it depends only on `@tanstack/query-core` not `@tanstack/react-query`.)

So, in either case, it's suboptimal.

**Comment by sandrosc** [comment]
Thank you for the fast answer! So the best solution for now might be doing a `window.reload()` to cause a refetch of the entire page?

**Comment by sandrosc** [comment]
You mean like an dummy atom around the query atom that catches the error if it occurs?

**Comment by sandrosc** [comment]
Another question, do you think this will work in the long run?

**Comment by dai-shi** [comment]
> seems to work by changing the user state to a different view where the query doesn't fail.

By dummy atom, I mean something like the user state ☝️ .

> do you think this library will work with the error boundary reset in the long run?

If it's about how Jotai handles error boundary, the functionality should be stable.
If it's about something very specific to this library, not sure. But, I think the current `dataAtom` is fairly a general Jotai atom.

That said, I agree there's some uncertainty, so using `statusAtom` would be a safer bet.

**Comment by sandrosc** [comment]
> If it's about something very specific to this library,

Yes I think it would be nice to integrate jotai and tanstack-query so that the case at the beginning of the issue can be handled easily. Handling user state separately defeats the purpose of the error boundary, which is supposed to be generic in my case.

**Comment by dai-shi** [comment]
I thought `dataAtom` and `statusAtom` could cover most cases, but they are still suboptimal.

We need maintainers with good understanding with tanstack/query-core and Jotai to revisit the API. (`jotai-urql` recently got a new maintainer and has a new API, dropping `atomsWithQuery`.)

Any volunteers?
